Source/Details/ASCollectionLayoutState.h

```diff
@@ -19,45 +19,86 @@
 #import <UIKit/UIKit.h>
 #import <AsyncDisplayKit/ASBaseDefines.h>
 
-@class ASElementMap, ASCollectionElement, ASLayout;
+@class ASCollectionLayoutContext, ASLayout, ASCollectionElement;
 
 NS_ASSUME_NONNULL_BEGIN
 
+@interface NSMapTable (ASCollectionLayoutConvenience)
+
++ (NSMapTable<ASCollectionElement *, UICollectionViewLayoutAttributes *> *)elementToLayoutAttributesTable;
+
+@end
+
 AS_SUBCLASSING_RESTRICTED
 @interface ASCollectionLayoutState : NSObject
 
-/// The elements used to calculate this object
-@property (nonatomic, strong, readonly) ASElementMap *elements;
+/// The context used to calculate this object
+@property (nonatomic, strong, readonly) ASCollectionLayoutContext *context;
 
+/// The final content size of the collection's layout
 @property (nonatomic, assign, readonly) CGSize contentSize;
 
-/// Element to layout attributes map. Should use weak pointers for elements.
-@property (nonatomic, strong, readonly) NSMapTable<ASCollectionElement *, UICollectionViewLayoutAttributes *> *elementToLayoutArrtibutesMap;
+/// The final content rect calculated from content size
+@property (nonatomic, assign, readonly) CGRect contentRect;
 
 - (instancetype)init __unavailable;
 
 /**
  * Designated initializer.
  *
- * @param elements The elements used to calculate this object
+ * @param context The context used to calculate this object
  *
  * @param contentSize The content size of the collection's layout
  *
- * @param elementToLayoutArrtibutesMap Map between elements to their layout attributes. The map may contain all elements, or a subset of them and will be updated later. 
- * Also, it should have NSMapTableObjectPointerPersonality and NSMapTableWeakMemory as key options.
+ * @param table A map between elements to their layout attributes. It may contain all elements, or a subset of them that will be updated later.
+ * It should be initialized using +[NSMapTable elementToLayoutAttributesTable] convenience initializer.
  */
-- (instancetype)initWithElements:(ASElementMap *)elements contentSize:(CGSize)contentSize elementToLayoutArrtibutesMap:(NSMapTable<ASCollectionElement *, UICollectionViewLayoutAttributes *> *)elementToLayoutArrtibutesMap NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithContext:(ASCollectionLayoutContext *)context contentSize:(CGSize)contentSize elementToLayoutAttributesTable:(NSMapTable<ASCollectionElement *, UICollectionViewLayoutAttributes *> *)table NS_DESIGNATED_INITIALIZER;
 
 /**
  * Convenience initializer.
  *
- * @param elements The elements used to calculate this object
+ * @param context The context used to calculate this object
+ *
+ * @param layout The layout describes size and position of all elements, or a subset of them and will be updated over time.
+ *
+ */
+- (instancetype)initWithContext:(ASCollectionLayoutContext *)context layout:(ASLayout *)layout;
+
+/**
+ * Returns all layout attributes present in this object.
+ */
+- (NSArray<UICollectionViewLayoutAttributes *> *)allLayoutAttributes;
```

Comment:
Looks like this isn't currently used. Unless you have a specific use in mind, let's remove for now.